### ISSUE TITLE
eth_signTypedData(): callback function is no longer called in v 4.9.0 on 'SIGN' button

### ISSUE THREAD

**Issue by Maxxi** [MetaMask/metamask-extension#4975]
The following code worked fine in <=v 4.8.0:

```
web3.currentProvider.sendAsync( {
            'eth_signTypedData',
            params,
            address
        }, 
        (err, result) => {
            // callback code
        }
);
```
But it stopped working in version 4.9.0.         
_(err, result) => {}_ 
callback function is not called anymore after confirming Signature Request (SIGN button)

However, the callback function is still called on 'CANCEL' button as before.

**Comment by Netherdrake** [comment]
+1

**Comment by nnkken** [comment]
+1
BTW, when will the current `eth_signTypedData` be deprecated and replaced by the merged EIP-712 behaviour?

**Comment by turshija** [comment]
I can confirm this too ...
When user clicks cancel - callback is called with first parameter null and second one contains this object:
```
{ 
  "error": {
    "code": -32603, 
    "message": "Error: MetaMask Message Signature: User denied message signature."
  },
  "id": undefined,
  "jsonrpc": undefined
}
```

But when user clicks "Sign" - nothing happens - callback doesn't trigger
It works in 4.8.0, but stops working in 4.9.0

**Comment by bitpshr** [comment]
Hi @Maxxi. This was caused by a premature version bump and will be fixed by #4982. We plan to do a patch release shortly thereafter. Thanks for the issue report and for your patience!

**Comment by Maxxi** [comment]
Hi @bitpshr . Thanks for your quick reaction on that problem.
